drivers/volume/portworx/portworx.go

```diff
@@ -1517,6 +1550,60 @@ func (d *portworx) getKvdbMembers(n node.Node) (map[string]metadataNode, error)
 	return kvdbMembers, err
 }
 
+func (d *portworx) CollectDiags(n node.Node) error {
+	var err error
+
+	pxNode, err := d.getPxNode(n, nil)
+	if err != nil {
+		return err
+	}
+
+	logrus.Debugf("Collecting diags on node %v", pxNode.Hostname)
+
+	if pxNode.Status == api.Status_STATUS_OFFLINE {
+		logrus.Debugf("Node %v is offline, collecting diags using pxctl", pxNode.Hostname)
+
+		opts := node.ConnectionOpts{
+			IgnoreError:     false,
+			TimeBeforeRetry: defaultRetryInterval,
+			Timeout:         defaultTimeout,
+		}
+
+		// Only way to collect diags when PX is offline is using pxctl
+		out, err := d.nodeDriver.RunCommand(n, "/opt/pwx/bin/pxctl sv diags -a -f", opts)
+		if err != nil {
+			return fmt.Errorf("Failed to collect diags on node %v, Err: %v %v", pxNode.Hostname, err, out)
+		}
+		logrus.Debugf("Successfully collected diags on node %v", pxNode.Hostname)
+		return nil
+	}
+
+	url := fmt.Sprintf("http://%s:9014", n.Addresses[0])
```

Comment:
it should be in the else statement, otherwise if pxNode.Status == api.Status_STATUS_OFFLINE then it will try to collect diags from http request.

Reply:
It will return after successfully collecting diags and will not try continue down to use http to collect diags as well